### ISSUE TITLE
Not showing actual memory

### ISSUE THREAD

**Issue by bannsec** [radare/radare2#8852]
This one hurt. Spent probably an hour or so figuring i was going crazy, but it turns out radare is simply not reporting the memory.

See the following binary: 
[bin.zip](https://github.com/radare/radare2/files/1484954/bin.zip)

Basically, set a break point after `sym.num` executes. It will be read into a static buffer at 0x804A040. Here's what radare tells me after checking the memory:

```
[0x080486d0]> px @ 0x0804a040
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x0804a040  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a050  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a060  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a070  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a080  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a090  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0a0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0b0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0c0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0d0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0e0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a0f0  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a100  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a110  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a120  0000 0000 0000 0000 0000 0000 0000 0000  ................
0x0804a130  0000 0000 0000 0000 0000 0000 0000 0000  ................
```

However, my input is indeed in that memory. The binary will execute it correctly, IDA shows it, but radare just tells me nothing is there.

**Comment by condret** [comment]
problem in section order passed from bin api i guess

**Comment by radare** [comment]
no. that's completelly correct. IDA lies, and you are using the debugger in a wrong way.

**Comment by radare** [comment]
this address corresponds to the .bss which is UNINITIALIZED MEMORY, set by the linker to 0, which is what you see in r2. Using gdb you also get 0 if you break in main, so that's expected.

**Comment by bannsec** [comment]
How so? Ida is showing me the same information the program itself is using.
If radare were correct a different execution path would happen.
Specifically, always return zero.

**Comment by bannsec** [comment]
I stated that I'm not breaking at main, however. I'm breaking after giving
it input.

**Comment by bannsec** [comment]
Effectively, the program reads in data to that memory region. After reading
my data, I ask radare to show me that memory region that contains my dat .
Radare says there's nothing there. However, continuing execution the
program behaves correctly as my data is indeed in the buffer. Gdb agrees
with Ida. Radare is the one that's off.

**Comment by radare** [comment]
Where do u set the bp to confirm that?

Can you paste the commands you r using?

**Comment by bannsec** [comment]
Do ya one better. I've documented the feature in an asciinema.

https://asciinema.org/a/148493

**Comment by radare** [comment]
Dont use -AA and dont use -d and it will work

**Comment by bannsec** [comment]
So testing those options, use or not use of -AA has no effect. However, opening the file directly with `-d` allows it to see the memory, where `ood` does not. My understanding is that `ood` was the interactive way of opening the file for debugging. I would not expect this to have any affect on successfully returning memory since that's a basic task of any debugger regardless of instantiation.

My new question is, is this still considered not a bug and if so, should people continue to utilize `ood` with the understanding that it might not return memory correctly?

**Comment by radare** [comment]
Yes it is a bug because in ood we should close previous maps . This is a regression in siol. Easy to fix. So nice to know the problem was in ood

**Comment by radare** [comment]
confirmed issue, only affecting linux, on mac all maps are closed, i dont use linux as main platform thats probably why this happened :P can anyone test on windows?

<img width="717" alt="screen shot 2017-11-22 at 12 50 48" src="https://user-images.githubusercontent.com/917142/33126160-d0b0f772-cf83-11e7-8857-63ef5b2d528f.png">

**Comment by radare** [comment]
should be fixed in 52145350d2225c91adc1a779c0f98c1ce92ec04a

thanks
